cmd/device.go

```diff
@@ -136,27 +141,31 @@ func runDevicePinger(wg *sync.WaitGroup, ping client.AgentPing, apiOrigin string
 			handleDeviceUpdate(ping, config)
 		}
 
-		if config.Enabled && config.Host != "" {
-			// ping server instead of sleeping
-			pinger, err := goping.NewPinger(config.Host)
-			if err == nil {
-				log.V(2).Info("Pinging server", "host", config.Host)
-				pinger.Count = AgentPingInterval * 10
-				pinger.Interval = time.Millisecond * 100
-				pinger.Timeout = time.Second * AgentPingInterval
-				pinger.Run()
-				log.V(1).Info("Done pinging server", "host", config.Host, "stats", *pinger.Statistics())
-				getPingStats(&ping, pinger.Statistics())
-			} else {
-				log.Error(err, "Failed to create pinger")
-				// sleep in between pings
-				time.Sleep(time.Second * AgentPingInterval)
-				getPingStats(&ping, nil)
-			}
+		if config.Enabled == false || config.Host == "" {
+			// There is no host configured, so move on
+			time.Sleep(AgentPingInterval)
+			getPingStats(&ping)
```

Comment:
Is it possible to simplify the branching here? It seems like both conditionals end up calling this sleep + getPingStats function. It might be cleaner to move this outside the if-statements but still inside the while loop.